server/server_test.go

```diff
@@ -430,17 +424,17 @@ func TestDisconnectingServer(t *testing.T) {
 		return
 	}
 
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(idleTimeout)
 	_, err = conn.Write([]byte("GET HTTP/1.1\r\n\r\n"))
 	if !assert.NoError(t, err) {
 		return
 	}
 
 	out, err := ioutil.ReadAll(conn)
-	if !assert.NoError(t, err) {
-		return
+	if err == nil {
```

Comment:
`assert.NoError` will fail if there's an error. In this particular case, we don't want to fail if there's an error, as it's possible that we'll see a connection reset (which is okay).